pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -207,6 +208,11 @@ func (am *Alertmanager) StopAndWait() error {
 }
 
 func (am *Alertmanager) SaveAndApplyConfig(cfg *apimodels.PostableUserConfig) error {
+	err := cfg.EncryptSecureSettings()
```

Comment:
I think this should be called where this `cfg` is being created at API level and not here. Alertmanager should not have to worry when to encrypt, and expect and pre-encrypted config.

Reply:
The best it would be to encrypt when marshalling the configuration but it should be done once and unfortunately we encode the configuration several times before storing to the database.
I believe that since the alert manager has the responsibility to save it to the database it does not harm to keep it here. But I'll try to move it in the API handler.